homeassistant/components/reolink/config_flow.py

```diff
@@ -96,7 +97,54 @@ async def async_step_reauth_confirm(
     async def async_step_dhcp(self, discovery_info: dhcp.DhcpServiceInfo) -> FlowResult:
         """Handle discovery via dhcp."""
         mac_address = format_mac(discovery_info.macaddress)
-        await self.async_set_unique_id(mac_address)
+        existing_entry = await self.async_set_unique_id(mac_address)
+        if (
+            existing_entry
+            and CONF_PASSWORD in existing_entry.data
+            and existing_entry.data[CONF_HOST] != discovery_info.ip
+        ):
+            # check the existing entry indeed has a connection problem
+            reolink_data: ReolinkData | None = self.hass.data.get(DOMAIN, {}).get(
+                existing_entry.entry_id
+            )
```

Comment:
It would be nice if there was a function to encapsulate this call so we aren't accessing `hass.data` directly here. (I realize its unavoidable but it would be a least nice to abstract it away)

Reply:
I split out this part into a `has_connection_problem` function.
I hope that is what you ment?